### PR TITLE
enabled backspace functionality in text-assessments

### PR DESCRIPTION
**Pull request by FuchsDominik**
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.

### Motivation and Context

It should be possible to refactor ones score in a text submission using backspaces.

### Description

removed a setting which prevented refactoring via backspace.

### Steps for Testing

In a text submission, when assessing a specific text fragment, it is now possible to remove entered credits through backspace.
If no credits are inserted, if will automatically insert 0.

Capture
Capture1

### CODE REVIEW

File: src/main/webapp/app/exercises/text/assess-new/textblock-feedback-editor/textblock-feedback-editor.component.html [hunk 1]
```diff
@@ -32,7 +32,6 @@
             [(ngModel)]="feedback.credits"
             id="score-{{ textBlock.id }}"
             (keydown.escape)="escKeyup()"
-            (keydown.backspace)="$event.preventDefault()"
```
Comment (jpbernius) [inline src/main/webapp/app/exercises/text/assess-new/textblock-feedback-editor/textblock-feedback-editor.component.html:35]: what about setting the credits to 0?

```ts
(keydown.backspace)="$event.preventDefault(); feedback.credits = 0"
```

**Comment by FuchsDominik** [comment]
Regarding the error: Leaving the credits empty leads to an error, however, removing credits through the _Entf_ key can also leave a empty credits field. Setting the score to 0 after clicking the backspace key does not fix the _Entf_ issue, so I should implement some sort of credit check.

**Comment by jpbernius** [comment]
Maybe try to set it to `0` if the value changes to nothing?

**Comment by FuchsDominik** [comment]
Update: Did a check so when a assessment is saved it checks whether credits are null and if, inserts 0.0 instead. This should fix the error of blank credits.

**Comment by jpbernius** [comment]
Why not set this client-side?

**Comment by csarpkaya** [comment]
I've tested the features on the specified test server with both instructor & tutor roles.

When I tried using Backspace to delete a given score, it didn't work. However Ctrl + Backspace does the job.

"Entf" button only deletes the credit if the given credit is already already highlighted. (meaning selected with mouse or ctrl+arrow keys)

I cannot submit an assessment that features a feedback without a score, I have to insert 0 so that it's valid.

**Comment by FuchsDominik** [comment]
@csarpkaya Can you specify what exactly doesnt work when trying to use backspace? I just deployed in again and tested it and backspace works as well as submitting without credits works.

**Comment by florian-glombik** [comment]
I am also not able to reproduce the problems addressed by @csarpkaya. _(I've used chrome and firefox)_

- "entf"/delete: worked fine for me, deleted the next symbol to the right _(if there is no symbol on the right side of the cursor nothing happens, but I think this is the expected behaviour - did this maybe lead to some confusion?)_
- backspace / control + backspace also worked fine for me
- saving an empty score field / entering non numeric values in the score field resulted in 0 points, as expected

**Comment by FuchsDominik** [comment]
@florian-glombik We solved the problem. Can forgot to deploy on the server, so the features where not up to date. After deploying again, he mentioned it works properly.

**Pull request merged**